app/controllers/bookingsync_application/webhooks/base_controller.rb

```diff
@@ -0,0 +1,31 @@
+class Webhooks::BaseController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  before_action :verify_signature
+
+  protected
+
+  def invalidate_request_and_log(message)
+    logger.info "Failing with 422 because: #{message}"
+    head 422
+  end
+
+  def verify_signature
+    Rails.logger.debug "bookingsync_hook_raw_post: #{request.raw_post}"
+    unless signature == request.headers["X-Content-Signature"]
+      invalidate_request_and_log "Bad BookingSync signature"
+    end
+  end
+
+  def signature
```

Comment:
I think it's best to have the first version released handling both, much easier to ensure everyone is up to speed before we migrate the signature. It's super simple as well. Just need to use http://ruby-doc.org/stdlib-2.3.0/libdoc/base64/rdoc/Base64.html#method-i-strict_encode64 for the new signature.

:+1: For dummy app testing